### PR TITLE
feature/rev date

### PR DESCRIPTION
**Pull request by martinyde**
https://jira.itkdev.dk/browse/LOOP-745
- LOOP-745: Added revision date
- LOOP-745: Add revision date module
- LOOP-745: Added config
- LOOP-745: Added date popup module

### CODE REVIEW

File: web/profiles/custom/os2loop/modules/os2loop_revision_date/config/install/system.action.extend_revision_date_one_month_action.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+status: true
+dependencies:
+  module:
+    - os2loop_revision_date
+id: extend_revision_date_one_month_action
+label: 'Set next revision date to 1 month from now'
+type: node
+plugin: os2loop_revision_date_set_action
+configuration:
+  time_span: 2629743
```
Comment (rimi-itk) [inline web/profiles/custom/os2loop/modules/os2loop_revision_date/config/install/system.action.extend_revision_date_one_month_action.yml:11]: We usually don't put config into a modules `config/install` folder, but keep it all in `config/sync` (for now).

File: config/sync/system.action.extend_revision_date_three_months_action.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  module:
+    - os2loop_revision_date
+_core:
+  default_config_hash: U5PfDTS42RKqbYCR30JF8XSIP2kAFizgQp5zCjpD4Mg
+id: extend_revision_date_three_months_action
+label: 'Set next revision date to 3 months from now'
+type: node
+plugin: os2loop_revision_date_set_action
+configuration:
+  time_span: three_months
```
Comment (rimi-itk) [inline config/sync/system.action.extend_revision_date_three_months_action.yml:14]: The value `'three_months'` will probably not work with `$this->timeService->getCurrentTime() + $this->getTimeSpan()` in `SetRevisionDate::execute`. Should it be numeric?

File: web/profiles/custom/os2loop/modules/os2loop_revision_date/src/Plugin/Action/SetRevisionDate.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  public function execute($node = NULL) {
+    if ($node) {
+      /** @var \Drupal\node\NodeInterface $node */
+      if (!$node->hasField('os2loop_shared_rev_date')) {
+        throw new \RuntimeException("Revisioning date field not found on node.");
+      }
+      $node->set('os2loop_shared_rev_date', date('Y-m-d', $this->timeService->getCurrentTime() + $this->getTimeSpan()));
```
Comment (rimi-itk) [inline web/profiles/custom/os2loop/modules/os2loop_revision_date/src/Plugin/Action/SetRevisionDate.php:68]: Can we use a `DrupalDateTime` as the value? If so, we can use relative times in its constructor, e.g.
```php
new DrupalDateTime('+1 month')
```
(where `'+1 month'` can be the `time_span` config value).

**Review by rimi-itk — CHANGES_REQUESTED** [review]
Some suggestions added.

File: web/profiles/custom/os2loop/modules/os2loop_revision_date/config/install/system.action.extend_revision_date_one_month_action.yml (hunk 1 above)
Comment (martinyde) [reply web/profiles/custom/os2loop/modules/os2loop_revision_date/config/install/system.action.extend_revision_date_one_month_action.yml]: Removed module config

File: config/sync/system.action.extend_revision_date_three_months_action.yml (hunk 2 above)
Comment (martinyde) [reply config/sync/system.action.extend_revision_date_three_months_action.yml]: Changed to relative strings.

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**